GitUI/CommandsDialogs/SettingsDialog/Pages/ChecklistSettingsPage.cs

```diff
@@ -221,54 +219,13 @@ private void GitBinFound_Click(object sender, EventArgs e)
 
         private void ShellExtensionsRegistered_Click(object sender, EventArgs e)
         {
-            string path = Path.Combine(AppSettings.GetInstallDir(), CommonLogic.GitExtensionsShellEx32Name);
-
-            if (!File.Exists(path))
+            try
             {
-                path = Assembly.GetAssembly(GetType()).Location;
-                path = Path.GetDirectoryName(path);
-                path = Path.Combine(path, CommonLogic.GitExtensionsShellEx32Name);
+                ShellExtensionManager.Register();
             }
-
-            if (File.Exists(path))
-            {
-                try
-                {
-                    ProcessStartInfo pi = new()
-                    {
-                        FileName = "regsvr32",
-                        Arguments = path.Quote(),
-                        Verb = "RunAs",
-                        UseShellExecute = true
-                    };
-
-                    var process = Process.Start(pi);
-                    process.WaitForExit();
-
-                    if (IntPtr.Size == 8)
-                    {
-                        path = path.Replace(CommonLogic.GitExtensionsShellEx32Name, CommonLogic.GitExtensionsShellEx64Name);
-                        if (File.Exists(path))
-                        {
-                            pi.Arguments = path.Quote();
-
-                            var process64 = Process.Start(pi);
-                            process64.WaitForExit();
-                        }
-                        else
-                        {
-                            MessageBox.Show(this, string.Format(_cantRegisterShellExtension.Text, CommonLogic.GitExtensionsShellEx64Name), TranslatedStrings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
-                        }
-                    }
-                }
-                catch (System.ComponentModel.Win32Exception)
-                {
-                    // User cancel operation, continue;
-                }
-            }
-            else
+            catch (Exception ex)
             {
-                MessageBox.Show(this, string.Format(_cantRegisterShellExtension.Text, CommonLogic.GitExtensionsShellEx32Name), TranslatedStrings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
+                MessageBoxes.ShowShellExtensionRegistrationError(ParentForm, ex);
```

Comment:
Would could probably use UserExternal exception and remove the custom message box.

Reply:
Done and updated the PR description.